store/src/main/java/com/dropbox/android/external/store4/StoreResponse.kt

```diff
@@ -98,6 +103,7 @@ sealed class StoreResponse<out T> {
     internal fun <R> swapType(): StoreResponse<R> = when (this) {
         is Error -> this as Error<R>
         is Loading -> this as Loading<R>
+        is NoNewData -> this as NoNewData<R>
```

Comment:
can we call this Empty or Absent instead of NoNewData ?

Reply:
actually reading more i guess this is different than absent :/.
but it is a bit confusing to send this down to the user. should this be instead in Fetcher response and not in StoreResponse so that it can only come out of fetcher?

Reply:
maybe it can be named NotModified (to match 304). I don't like it much either so i don't have a strong opinion either.

Reply:
The reason we need to expose this to the user is that currently, if the fetceh returns an empty flow we will emit `Loading` as the last value. `NoNewData` is the most neutral way I can map an empty flow to something returned to the user. `Empty`,`Absent` and `NotModified` all have a more nuanced meaning than the generic "nothing new was returned".

with this change we will now emit the following - 

for stream(Request.cached(key, refresh=true)) we will return:
Cache
Source of truth
Fetcher - > Loading
Fetcher - > NoNewData
(future Source of truth updates)

for stream(Request.fresh(key)) we will return:
Fetcher - > Loading
Fetcher - > NoNewData
Cache // open to removing this. for SoT, but needed for when we only have network so went with consistency) 
Source of truth
(future Source of truth updates)

Reply:
yea i think this is the right option if we are falling back to local data, it is better to go through cache.

it is a bit unfortunate that skipCache returns data from cache but it is realistic.

can you add this comment into the code of `stream` so that next person hitting this will realize it is very intentional.

Reply:
added a comment to `fresh`